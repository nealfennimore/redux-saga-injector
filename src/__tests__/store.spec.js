import augmentStore from '../store';

describe( 'Store augmentation', ()=>{
    let store, sagaMiddleware;
    beforeEach( ()=>{
        store = {
            dispatch: jest.fn()
        };
        sagaMiddleware = {
            run: jest.fn()
        };
        augmentStore( sagaMiddleware, store );
    } );

    test( 'store close should end all running sagas', ()=>{
        store.close();
        expect( store.dispatch.mock.calls ).toMatchSnapshot();
    } );
    test( 'should alias sagaMiddleware run', ()=>{
        expect( store.runSaga ).toEqual( sagaMiddleware.run );
    } );
} );
