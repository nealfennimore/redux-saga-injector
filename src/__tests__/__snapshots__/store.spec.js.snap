// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store augmentation store close should end all running sagas 1`] = `
Array [
  Array [
    Object {
      "type": "@@redux-saga/CHANNEL_END",
    },
  ],
]
`;
